mycpp/mylib.cc

```diff
@@ -208,6 +208,45 @@ Str* Str::join(List<Str*>* items) {
   return new Str(result, len);
 }
 
+
+  Str* Str::upper() {
+    Str* self = this;
+    Str* result = 0;
+
+    /* StackRoots stroots = {&result, &self}; */
+
+    result = mylib::BlankStr(self->len_);
+
+    // TODO(Jesse): Have to cast off const-ness .. this is intolerable in the
+    // long-run.  How should we deal?
+    char *buffer = (char*)result->data_;
+
+    for (int char_index = 0; char_index <= self->len_; ++char_index)
+    {
+      buffer[char_index] = toupper(self->data_[char_index]);
+    }
+    return result;
+  }
+
+  Str* Str::lower() {
+    Str* self = this;
```

Comment:
`mylib.cc` are the leaky implementations, so they don't need any StackRoots or `self` stuff.  This is just plain C++ code that leaks

I started the first GC bindings which is `core/qsn_qsn.h` -- there we have to take care.

And I started adding unit tests in `cpp/gc_bindings_test`, those are an absolute MUST for the GC code because of the potential for GC bugs.

So 

- the leaky code is `mylib.{h,cc}`
- the GC code is `{gc_heap,mylib2,my_runtime}.{h,cc}`

I'm thinking of renaming it to `mylib.leaky.h` and mylib.h` so it's clear.  We can name things so the goal is to get rid of everything "leaky"

Reply:
Yeah, I was thinking I'd just copy the implementation to the GC'd version of `Str` when the time came, and it would be nice to just uncomment a line to have it work .. but it does add a fair bit of noise.  I'll take that stuff out.

I do like the idea of unifying the naming a bit more, though I'm touching all of those files a lot right now so can we hold off on doing that till my current PRs are closed?  Wanna avoid a conflict-induced headache ;)